freqtrade/resolvers/strategy_resolver.py

```diff
@@ -24,8 +24,6 @@ class StrategyResolver(IResolver):
     This class contains all the logic to load custom strategy class
     """
 
-    __slots__ = ['strategy']
```

Comment:
Would you be able to explain what errors you encounter (in which circumstances) due to this?

I'd like to understand what errors you're encountering, since i've never seen a problem due to this.

Reply:
In a Jupyter notebook, including `__slots__ = ['strategy']` results in `TypeError`: 

```
from freqtrade.plot.plotting import load_and_plot_trades
load_and_plot_trades(config)
2019-09-04 20:33:59,904 - freqtrade.resolvers.iresolver - INFO - Using resolved strategy NewStrategy from '/Users/jonathanraviotta/Documents/Repos/freqtrade/user_data/user_repo/strategies/new_strategy.py'...
 TypeError: descriptor 'strategy' for 'StrategyResolver' objects doesn't apply to 'StrategyResolver' object 
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-25-0263847289e7> in <module>
      1 from freqtrade.plot.plotting import load_and_plot_trades
----> 2 load_and_plot_trades(config)

~/Documents/Repos/freqtrade/freqtrade/plot/plotting.py in load_and_plot_trades(config)
    339     :return: None
    340     """
--> 341     strategy = StrategyResolver(config).strategy
    342 
    343     plot_elements = init_plotscript(config)

~/Documents/Repos/freqtrade/freqtrade/resolvers/strategy_resolver.py in __init__(self, config)
     38         self.strategy: IStrategy = self._load_strategy(strategy_name,
     39                                                        config=config,
---> 40                                                        extra_dir=config.get('strategy_path'))
     41 
     42         # make sure experimental dict is available

TypeError: descriptor 'strategy' for 'StrategyResolver' objects doesn't apply to 'StrategyResolver' object
```